src/components/graph-view.js

```diff
@@ -558,7 +558,7 @@ class GraphView extends React.Component<IGraphViewProps, IGraphViewState> {
 
     onSelectNode && onSelectNode(null);
     onSelectEdge && onSelectEdge(null);
-    onSelect && onSelect({ nodes: [], edges: [] });
+    onSelect && onSelect({ nodes: new Set(), edges: new Set() });
```

Comment:
Sets have a O(1) complexity for searches, so this speeds up our checks when trying to determine if a node or edge is within the selected sets of nodes or edges.

Reply:
Curious, which areas did a map add complexity? (We just discussed moving toward a map/dictionary in the next major version, so using a map would be more inline with that plan.)

Reply:
When we select nodes we get an array of nodes, so then we need to convert it to something that has a O(1) time complexity for searches (when moving nodes we have to determine if you're touching one that's in a selection set). 

There are some options: 

- Set: O(1) search time, array-like, can be instantiated from an array. Downsides: searches require exact matches, can't do object property matches or non-referential matches. This is OK in our case because the selected nodes/edges are never supposed to be copied to a new reference.
- Map: O(1) search time, object/dictionary-like, can reference nodes and edges by ID rather than object references, has a size property. Downsides: can't be directly instantiated from an array, need to add individual key/value pairs in a loop.
- Array: Slow search time when moving the mouse, blocks rendering.
- Primitive Object: Like Map. Slightly easier than Map to understand for Javascript developers because they're used to `{}` bracket notation. Downsides: similar downside to Map, can't directly translate arrays to objects without looping the key/value pairs.

In the 8.0 version we'll convert the input nodes and edges to a Map (not {}), because it gives us the size property and has the benefits of primitive objects.